### DISCUSSION TITLE
NANO ESP32 with Adafruit HX3857 3.5 TFT

### DISCUSSION THREAD

**Discussion by jgadelman06** [Bodmer/TFT_eSPI#2999] [Compatible displays and setup files]
I can't get my NANO ESP32 to work with my Adafruit 3.5 TFT.  All I get is a white screen. It works fine with Adafruit libraries.

**Driver I uncommented:**
#define HX8357D_DRIVER

**Pin setup I uncommented:**
// ###### EDIT THE PIN NUMBERS IN THE LINES FOLLOWING TO SUIT YOUR ESP32 SETUP   ######
// For ESP32 Dev board (only tested with ILI9341 display)
// The hardware SPI can be mapped to any pins
#define TFT_MISO 47
#define TFT_MOSI 38
#define TFT_SCLK 48
#define TFT_CS   21 // Chip select control pin
#define TFT_DC   18 // Data Command control pin
//#define TFT_RST  17  // Reset pin (could connect to RST pin)
#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

**Read_User_Setup returned the following:**
[code]
TFT_eSPI ver = 2.5.34
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
Display driver = 8357
Display width  = 320
Display height = 480

MOSI    = GPIO 38
MISO    = GPIO 47
SCK     = GPIO 48
TFT_CS   = GPIO 21
TFT_DC   = GPIO 18

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]

I also tried calling pins out as D12, D11, D13, D10, D9, as well as 12, 11, 13, 10, 9.
I have no clue what I'm doing wrong. Any help would be much appreciated.

**Comment by lowki** [comment]
I have the same issue with an Adafruit ST7789 1.54 screen.
Anyone ?
https://github.com/Bodmer/TFT_eSPI/discussions/2991

**Comment by Bodmer** [comment]
This may help for the interface connections:
https://m.youtube.com/watch?v=f1WtrL8OZVo

**Reply by jgadelman06** [reply]
Thanks for responding. The video gave me some new ideas, and I finally figured it out after several hours of tinkering. In the IDE, I had to go to tools>pin numbering, and select "by GPIO."
